console/console-init/ui/src/modules/iot-device-detail/components/ConfigurationInfo/ConfigurationInfo.tsx

```diff
@@ -12,64 +12,181 @@ import {
   Card,
   CardBody,
   Title,
-  Alert
+  DropdownPosition,
+  Tooltip,
+  TooltipPosition,
+  Button,
+  ButtonVariant
 } from "@patternfly/react-core";
+import { FilterIcon, ExclamationCircleIcon } from "@patternfly/react-icons";
+import { StyleSheet } from "@patternfly/react-styles";
 import { AdapterList, IAdapterListProps } from "components";
 import {
   CredentialsView,
   ICredentialsViewProps
 } from "modules/iot-device-detail/components";
+import { DropdownWithToggle, IDropdownOption } from "components";
+import {
+  credentialsTypeOptions,
+  getDefaultCredentialsFiterOption
+} from "modules/iot-device-detail/utils";
+import { CredentialsType } from "constant";
+
+const style = StyleSheet.create({
+  filter_dropdown: {
+    paddingLeft: 20
+  }
+});
 
 export interface IConfigurationInfoProps
   extends Pick<IAdapterListProps, "adapters">,
     Pick<ICredentialsViewProps, "credentials"> {
   id: string;
+  onSelectFilterType?: (value: string) => void;
+  onSelectFilterValue?: (value: string) => void;
 }
 
 export const ConfigurationInfo: React.FC<IConfigurationInfoProps> = ({
   id,
   adapters,
-  credentials
+  credentials,
+  onSelectFilterType,
+  onSelectFilterValue
 }) => {
+  const [credentialType, setCredentialType] = useState<string>("all");
+  const [filterOptions, setFilterOptions] = useState<IDropdownOption[]>([]);
+  const [selectedFilterValue, setSelectedFilterValue] = useState();
```

Comment:
Is there any reason we are not passing a type here, I am getting TSLint warnings about incompatible types ( string | string[] ) and undefined. Same with the `filterType` and `filterValue` of ConfigurationInfoContainer.

Reply:
sorry, i don't get the issue. can you please send me some detail.